parastell/magnet_coils.py

```diff
@@ -268,11 +270,30 @@ def export_step(self, step_filename="magnet_set", export_dir=""):
             self.step_filename
         ).with_suffix(".step")
 
-        coil_set = cq.Compound.makeCompound(
-            [coil.solid for coil in self.magnet_coils]
-        )
+        coil_set = cq.Compound.makeCompound(self.coil_solids)
         cq.exporters.export(coil_set, str(export_path))
 
+    def extract_solids_and_mat_tag(self, solids, mat_tags):
+        """Appends magnet set CadQuery solid objects and material tag to
+        corresponding input lists.
+
+        Arguments:
+            solids (list): list to which magnet set CadQuery solid objects will
+                be appended.
+            mat_tags (list): list to which magnet set material tag will be
+                appended. Note that the material tag will be appended once for
+                each magnet solid.
+
+        Returns:
+            solids (list): updated list including magnet set CadQuery solid
+                objects.
+            mat_tags (list): updated list including magnet set material tags.
+        """
+        solids.extend(self.coil_solids)
+        [mat_tags.append(self.mat_tag) for _ in self.coil_solids]
```

Comment:
This seems more readable to me
```suggestion
        mat_tags.extend([self.mat_tag] * len(self.coil_solids))
```